lib/redispub/publisher.go

```diff
@@ -63,26 +63,33 @@ var metricLastCommandDuration = promauto.NewGauge(prometheus.GaugeOpts{
 
 // PublishStream reads Publications from the given channel and publishes them
 // to Redis.
-func PublishStream(clients []redis.UniversalClient, in <-chan *Publication, opts *PublishOpts, stop <-chan bool) {
+func PublishStream(clients []redis.UniversalClient, in <-chan *Publication, opts *PublishOpts, stop <-chan bool, ordinal int) {
+	metricLastOplogEntryStaleness := promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "otr",
+		Subsystem: "redispub",
+		Name:      fmt.Sprintf("last_entry_staleness_seconds_%d", ordinal),
+		Help:      "Gauge recording the difference between this server's clock and the timestamp on the last published oplog entry.",
+	})
+
 	// Start up a background goroutine for periodically updating the last-processed
 	// timestamp
 	timestampC := make(chan primitive.Timestamp)
-	for _,client := range clients {
+	for _, client := range clients {
 		go periodicallyUpdateTimestamp(client, timestampC, opts)
```

Comment:
Does this work correctly with the parallelized write loop? I think we want to remember the last-seen timestamp for each write loop, then checkpoint the lowest amongst them to redis?

Reply:
I opted for each write loop checkpointing separately, and then having the startup read them all. Do you think that's sufficient?

Reply:
Yeah, as long as the behavior is right, i.e., it resumes from the correct point such that none are missed